From 28578b12f731be014d3b09e78ee9aff95b46265e Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Cl=C3=A9ment=20P=C3=A9ron?= <peron.clem@gmail.com>
Date: Wed, 11 Mar 2020 21:11:16 +0100
Subject: [PATCH 4/7] arm64: dts: allwinner: Update Beelink GS1 regulator
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Beelink GS1 regulators have been copied from another H6 board.

Even if it's work properly set the regulator properly.

Signed-off-by: Clément Péron <peron.clem@gmail.com>
---
 .../dts/allwinner/sun50i-h6-beelink-gs1.dts   | 46 ++++++++++++-------
 1 file changed, 29 insertions(+), 17 deletions(-)

diff --git a/arch/arm64/boot/dts/allwinner/sun50i-h6-beelink-gs1.dts b/arch/arm64/boot/dts/allwinner/sun50i-h6-beelink-gs1.dts
index df6d872c34e2..77cea2bfb58f 100644
--- a/arch/arm64/boot/dts/allwinner/sun50i-h6-beelink-gs1.dts
+++ b/arch/arm64/boot/dts/allwinner/sun50i-h6-beelink-gs1.dts
@@ -51,6 +51,24 @@
 		regulator-always-on;
 	};
 
+	reg_vcc33wifi: vcc33wifi {
+		compatible = "regulator-fixed";
+		regulator-name = "vcc33-wifi";
+		regulator-min-microvolt = <3300000>;
+		regulator-max-microvolt = <3300000>;
+		vin-supply = <&reg_vcc5v>;
+		gpio = <&r_pio 0 10 GPIO_ACTIVE_HIGH>; /* PL10 */
+		enable-active-high;
+	};
+
+	reg_vccwifiio: vccwifiio {
+		compatible = "regulator-fixed";
+		regulator-name = "vcc-wifi-io";
+		regulator-min-microvolt = <3300000>;
+		regulator-max-microvolt = <3300000>;
+		vin-supply = <&reg_vcc33wifi>;
+	};
+
 	sound-spdif {
 		compatible = "simple-audio-card";
 		simple-audio-card,name = "sun50i-h6-spdif";
@@ -134,8 +152,9 @@
 };
 
 &pio {
+	vcc-pc-supply = <&reg_bldo2>;
 	vcc-pd-supply = <&reg_cldo1>;
-	vcc-pg-supply = <&reg_aldo1>;
+	vcc-pg-supply = <&reg_vccwifiio>;
 };
 
 &r_i2c {
@@ -163,35 +182,32 @@
 				regulator-always-on;
 				regulator-min-microvolt = <3300000>;
 				regulator-max-microvolt = <3300000>;
-				regulator-name = "vcc-pl";
+				regulator-name = "vcc-pl-led-ir";
 			};
 
 			reg_aldo2: aldo2 {
 				regulator-min-microvolt = <3300000>;
 				regulator-max-microvolt = <3300000>;
-				regulator-name = "vcc-ac200";
+				regulator-name = "vcc-audio-tv-ephy-mac";
 				regulator-enable-ramp-delay = <100000>;
 			};
 
 			reg_aldo3: aldo3 {
-				regulator-always-on;
-				regulator-min-microvolt = <3300000>;
-				regulator-max-microvolt = <3300000>;
-				regulator-name = "vcc25-dram";
+				/* unused */
 			};
 
 			reg_bldo1: bldo1 {
 				regulator-always-on;
 				regulator-min-microvolt = <1800000>;
 				regulator-max-microvolt = <1800000>;
-				regulator-name = "vcc-bias-pll";
+				regulator-name = "vcc-dram-bias-pll";
 			};
 
 			reg_bldo2: bldo2 {
 				regulator-always-on;
 				regulator-min-microvolt = <1800000>;
 				regulator-max-microvolt = <1800000>;
-				regulator-name = "vcc-efuse-pcie-hdmi-io";
+				regulator-name = "vcc-pcie-pc-efuse-hdmi";
 			};
 
 			reg_bldo3: bldo3 {
@@ -209,19 +225,15 @@
 				regulator-always-on;
 				regulator-min-microvolt = <3300000>;
 				regulator-max-microvolt = <3300000>;
-				regulator-name = "vcc-3v3";
+				regulator-name = "vcc-pd-io-emmc-nand-card-usb-uart";
 			};
 
 			reg_cldo2: cldo2 {
-				regulator-min-microvolt = <3300000>;
-				regulator-max-microvolt = <3300000>;
-				regulator-name = "vcc-wifi-1";
+				/* unused */
 			};
 
 			reg_cldo3: cldo3 {
-				regulator-min-microvolt = <3300000>;
-				regulator-max-microvolt = <3300000>;
-				regulator-name = "vcc-wifi-2";
+				/* unused */
 			};
 
 			reg_dcdca: dcdca {
@@ -272,7 +284,7 @@
 	 *
 	 * vcc-pl-supply = <&reg_aldo1>;
 	 */
-	vcc-pm-supply = <&reg_aldo1>;
+	vcc-pm-supply = <&reg_vccwifiio>;
 };
 
 &spdif {
-- 
2.20.1

